app/Domain/Mail/MailManager.php

```diff
@@ -147,4 +93,26 @@ private function getUniqueContactsWithPreferredContactType(Integration $integrat
 
         return $uniqueContacts;
     }
+
+    public function sendMail(MailEnabled $event, ?Template $template): ?Integration
```

Comment:
Inject the `$id` here so there is no need for the interface `MailEnabled`.

There is also no reason for `$template` to be nullable. That indicates a misconfiguration and should throw.